### PR TITLE
FunctionAppAuthenticationEnableRule: new azure rule

### PR DESCRIPTION
**Pull request by idan-aviv**

### CODE REVIEW

File: cloudrail/version.py [hunk 1]
```diff
@@ -1 +1 @@
-__version__ = 'v0.0.26'
+__version__ = 'v0.0.27-beta.4'
```
Comment (Imanuel-Mizrahi) [inline cloudrail/version.py:1]: as far as I know, we do not push beta, so that should be changed

**Review by Imanuel-Mizrahi — APPROVED** [review]

File: cloudrail/version.py (hunk 1 above)
Comment (idan-aviv) [reply cloudrail/version.py:1]: @Imanuel-Mizrahi i didn't change it, its automatically added by github flow when you deploy to pipy

File: cloudrail/knowledge/context/azure/azure_resources/web_app/azure_function_app.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from typing import Optional, List
+from cloudrail.knowledge.context.azure.azure_resources.azure_resource import AzureResource
+from cloudrail.knowledge.context.azure.azure_resources.constants.azure_resource_type import AzureResourceType
+from cloudrail.knowledge.context.azure.azure_resources.web_app.auth_settings import AuthSettings
+
+
+class AzureFunctionApp(AzureResource):
```
Comment (yi2020) [inline cloudrail/knowledge/context/azure/azure_resources/web_app/azure_function_app.py:7]: Missing docs

File: cloudrail/knowledge/rules/azure/non_context_aware/function_app_authentication_enable_rule.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class FunctionAppAuthenticationEnableRule(AzureBaseRule):
+
+    def get_id(self) -> str:
+        return 'non_car_authentication_enabled_in_function_app'
+
+    def execute(self, env_context: AzureEnvironmentContext, parameters: Dict[ParameterType, any]) -> List[Issue]:
+        issues: List[Issue] = []
+        for func_app in env_context.function_apps.values():
```
Comment (yi2020) [inline cloudrail/knowledge/rules/azure/non_context_aware/function_app_authentication_enable_rule.py:16]: Why do you need .values() ? When using the context in AWS, we don't need .values()

Comment (yi2020) [reply cloudrail/knowledge/rules/azure/non_context_aware/function_app_authentication_enable_rule.py:16]: (or in other words, why use AliasesDict)

File: cloudrail/knowledge/context/azure/azure_resources/web_app/azure_function_app.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
+from typing import Optional, List
```
Comment (yi2020) [inline cloudrail/knowledge/context/azure/azure_resources/web_app/azure_function_app.py:1]: Can you `git mv` everything under azure_resources to simply azure, so it matches the directory structure we have for `/context/aws`? I know it's not part of this rule, but I've already asked for this in another PR and it wasn't done.

**Review by yi2020 — CHANGES_REQUESTED** [review]

File: cloudrail/knowledge/rules/azure/non_context_aware/function_app_authentication_enable_rule.py (hunk 3 above)
Comment (Tomerva) [reply cloudrail/knowledge/rules/azure/non_context_aware/function_app_authentication_enable_rule.py:16]: keep using AliasesDict, theres just no need to do .values() when iterating

File: cloudrail/version.py (hunk 1 above)
Comment (Tomerva) [reply cloudrail/version.py:1]: when you do release, it automatically modifies version.py to include the new version

**Review by yi2020 — APPROVED** [review]

**Pull request merged**